onadata/apps/api/viewsets/project_viewset.py

```diff
@@ -226,8 +232,67 @@ def star(self, request, *args, **kwargs):
 
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @action(detail=True, methods=["GET", "POST"])
+    def invitations(self, request, *args, **kwargs):
+        """List, Create project invitations"""
+        project = self.get_object()
+        method = request.method.upper()
+
+        if method == "GET":
+            invitations = project.invitations.all()
+            invitation_status = request.query_params.get("status")
+
+            if invitation_status:
+                invitations = invitations.filter(status=invitation_status)
+
+            serializer = ProjectInvitationSerializer(invitations, many=True)
+            return Response(serializer.data)
+
+        if method == "POST":
+            serializer = ProjectInvitationSerializer(
+                data={**request.data, "project": project.pk}
+            )
+            serializer.is_valid(raise_exception=True)
+            email = request.data.get("email")
```

Comment:
Do we need to first check if email is in the payload?

Reply:
`serializer.is_valid(raise_exception=True)` is handling that. If the email is not in the payload a 400 error will be raised

Reply:
Maybe in that case I should use `serializer.data['email']` instead. I have updated the code